pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,241 @@
+from math import log
+import itertools
+import numpy as np
+from operator import sub
+from operator import add
+
+class MPLPAlg:
+
+    """
+    The MAP problem asks us to maximize some potentials. (call this primal).
```

Comment:
Add proper documentation. Please refer to any class in ```models``` module to get an idea of what are things needed to be specified in the documentation.